subsys/logging/log_backend_swo.c

```diff
@@ -36,7 +36,8 @@
 #if CONFIG_LOG_BACKEND_SWO_FREQ_HZ == 0
 #define SWO_FREQ_DIV  1
 #else
-#define SWO_FREQ (DT_CPU_CLOCK_FREQUENCY + (CONFIG_LOG_BACKEND_SWO_FREQ_HZ / 2))
+#define CPU_FREQ DT_PROP(DT_PATH(cpus, cpu_0), clock_frequency)
```

Comment:
In case `clock_frequency` property is missing the `DT_PROP` will generate a long and I assume somewhat cryptic compile time error message. I propose that we check for the existence of `clock_frequency` property and provide our own compile time error message if it's missing.

This issue will be especially visible on STM32 devices since they claim SWO support, however they generally don't define `clock_frequency` property. At least not the in-tree evaluation boards. The SWO works but only if `CONFIG_LOG_BACKEND_SWO_FREQ_HZ == 0` since then this line is not compiled. Any user trying to adjust the SWO speed will trigger the error.

Side remark: this incomplete support for SWO on STM32 devices is something we should solve in the future. There were some specific reasons why `clock_frequency` property was not defined on STM32, @erwango will know more.

Reply:
Any suggestion on the error to generate?

Reply:
I would put something simple like "Missing DT 'clock-frequency' property on cpu@0 node".

Reply:
Done.

Reply:
Our user friendly error message is printed, however the `DT_PROP` macro still generates a lot of spam. I propose to write it as
```
#if !DT_NODE_HAS_PROP(DT_PATH(cpus, cpu_0), clock_frequency)
#error "Missing DT 'clock-frequency' property on cpu@0 node"
#define CPU_FREQ 0
#else
#define CPU_FREQ DT_PROP(DT_PATH(cpus, cpu_0), clock_frequency)
#endif
```
I don't mean to be picky but this is user facing code, i.e. regular, inexperienced users may trigger the error.

Reply:
No worries.  I tweaked it so the error should be more sane.